### PR TITLE
feat: impl custom manifests output in the KCL UI and add more UI tests

### PR DESCRIPTION
**Pull request by Peefy**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

feat: #94 

* Add the UI impl of custom manifests output and more UI tests.

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

* unit test suites: test_custom_manifests_output at kclvm/runner/src/tests.rs

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by chai2010 — APPROVED** [review]
LGTM

File: kclvm/runner/src/custom_manifests_data/schema.stdout.golden [hunk 1]
```diff
@@ -0,0 +1,5 @@
+age: 1
```
Comment (Peefy) [inline kclvm/runner/src/custom_manifests_data/schema.stdout.golden:1]: Need to use serde_yaml needs expand Indexmap's YAML-ordered output capability. cc @NeverRaR

File: kclvm/runner/src/custom_manifests_data/dict_ignore_none.stdout.golden [hunk 2]
```diff
@@ -0,0 +1,4 @@
+age: 10
```
Comment (Peefy) [inline kclvm/runner/src/custom_manifests_data/dict_ignore_none.stdout.golden:1]: After YAML/JSON conversion, the results of the runtime plan lose the original storage order of KCL values, and the code of the runner needs to be reconstructed.

File: kclvm/runner/src/custom_manifests_data/schema.stdout.golden (hunk 1 above)
Comment (Peefy) [reply kclvm/runner/src/custom_manifests_data/schema.stdout.golden:1]: > Need to use serde_yaml needs expand Indexmap's YAML-ordered output capability. cc @NeverRaR

fixed in https://github.com/KusionStack/KCLVM/pull/309

**Review by zong-zhe — APPROVED** [review]
LGTM

**Review by NeverRaR — APPROVED** [review]
LGTM

File: kclvm/runner/src/tests.rs [hunk 3]
```diff
@@ -370,11 +372,16 @@ fn test_exec_file() {
     std::panic::set_hook(prev_hook);
 }
 
+fn test_custom_manifests_output() {
```
[the remark this answers is not in the document]
Comment (Peefy) [reply kclvm/runner/src/tests.rs:375]: > Need more UI tests on array value output.

Add more UI tests about list values

File: kclvm/runner/src/custom_manifests_data/dict_ignore_none.stdout.golden (hunk 2 above)
Comment (Peefy) [reply kclvm/runner/src/custom_manifests_data/dict_ignore_none.stdout.golden]: > After YAML/JSON conversion, the results of the runtime plan lose the original storage order of KCL values, and the code of the runner needs to be reconstructed.

fixed in #311

**Pull request merged**